pkg/infra/spot.go

```diff
@@ -110,31 +110,33 @@ var _ = Describe("Running on Spot", framework.LabelMachines, framework.LabelSpot
 			}
 			Expect(machineSetReady).To(BeTrue(), "Failed to create a spot backed MachineSet")
 		})
-	})
 
-	AfterEach(func() {
-		specReport := CurrentSpecReport()
-		if specReport.Failed() {
-			Expect(gatherer.WithSpecReport(specReport).GatherAll()).To(Succeed(), "StateGatherer should be able to gather resources")
-		}
+		DeferCleanup(func() {
```

Comment:
For the sake of making this maintainable, would it make sense to pair this with the line where we clear `delObjects` at the beginning of the before each? Do we risk otherwise not always clearing up if setup breaks mid way through?